primer/react
# fix(PageHeader): add role prop and aria-label in top-level element

Closes https://github.com/github/primer/issues/3466



### Changelog



#### New


- Adds `role` prop in `PageHeader` 
- Adds `PageHeader` tests to assert added functionality

#### Changed


- Adds `role="banner"` and `aria-label` to applicable instances of `PageHeader` in Storybook and tests
- Sets `aria-label` attribute in `PageHeader`'s "root" element mapped to the supplied `aria-label`
- Adjusts `PageHeader`'s `aria-label` docs description to reference "landmark" instead of inaccurate "main landmark"
- Includes `aria-label` prop in `PageHeader` Playground story

#### Removed


N/A

### Rollout strategy




### Testing & Reviewing



Integration Test: https://github.com/github/github/pull/341930

- Check [preview Playground story](https://primer-21157c34ef-13348165.drafts.github.io/storybook/?path=/story/components-pageheader--playground):
    - Notice in the DOM the "banner" role in the top-level `Box` element and the `aria-label` that maps to the `title` prop
    - Change `title` prop in playground and notice the `aria-label` change in the DOM as well
- Check [preview Default story](https://primer-21157c34ef-13348165.drafts.github.io/storybook/?path=/story/components-pageheader--default):
    - Notice in the DOM the "banner" role in the top-level `Box` element and the `aria-label`

### Merge checklist
